mandatory/3-playing-computer.js

```diff
@@ -7,36 +7,54 @@
   Answer the following questions:
 
   1. This program throws an error. Why? (If you can't find it, try executing it).
+  The variable b is missing.
+  
   2. Remove the line that throws the error.
+  Removed "console.log(b)"
+
   3. What is printed to the console?
+  2
+  6
+  4
+  9
+  6
+  13
+  8
+
   4. How many times is "f1" called?
+  2
+
   5. How many times is "f2" called?
+  3
+
   6. What value does the "a" parameter take in the first "f1" call?
+  8
```

Comment:
#### Declaration of `f1`

```js
const f1 = function (a, b) {
  return a + b;
};
```

We can see from the creation of `f1` that its first parameter is `a`. Therefore whenever `f1` is called, its **first argument** will be assigned to the **first parameter** `a`. Let's look at where `f1` is called:

#### Call site for `f1`

```js
} else {
  const e = f1(i, a);
  console.log(e);
}
```

`f1` is called when `i` is either `1` or `3`.

Therefore, the first time `f1` is called `i` must be equal to 1.

So `f1`'s first argument is 1 and therefore inside the function `f1`, `a` will be assigned a value of 1